src/Controller/CompanyController.php

```diff
@@ -32,210 +26,93 @@ class CompanyController
-	            'label'  => ' ',
-	            'attr'   =>  array(
-	                'class'   => 'input-field'),
-	            'constraints' => array(new Assert\NotBlank()),
-            ))
-            ->add('email', EmailType::class, array(
-	            'label'  => ' ',
-	            'attr'   =>  array('class'   => 'input-field'),
-	            'constraints' => array(new Assert\Email(array(
-		            'message' => 'The email "{{ value }}" is not a valid email.',
-		            'checkMX' => true,
-		        )), new Assert\NotBlank())
-            ))
-            ->add('category_id', ChoiceType::class, array(
-            	'choices' => array('Restaurant' => 1, 'Fast Food' => 2, 'Market' => 3, 'Drug Store' => 4, 'Other' => 5),
-            	'label'  => ' ',
-	            'attr'   =>  array(
-	                'class'   => 'select-field')
-            ))
-            ->add('delivery', CheckboxType::class, array(
-	            'label'  => ' ',
-	            'required' => false,
-	            'attr'   =>  array(
-	                'class'   => 'checkbox-field')
-            ))
-            ->add('radio_choice', ChoiceType::class, array (
-            	'choices' => array ('Choice A' => 1,
-					            	'Choice B' => 2,
-					            	'Choice C' => 3),
-            	'label'  => ' ',
-	            'attr'   =>  array(
-	                'class'   => 'radio-field')
-            ))
-            ->add('description', TextareaType::class, array(
-	            'label'  => ' ',
-	            'required' => false,
-	            'attr'   =>  array(
-	                'class'  => 'textarea-field'),
-	            'constraints' => new Assert\Length(array('max' => 255))
-            ))
-            ->add('FileUpload', FileType::class, array (
-            	'label' => ' '
-            ))
-            ->getForm();
-    	return $form;
-	}
+        if (null === $company) {
+            return new Response($app['twig']->render('errors/404.html.twig'));
+        }
+
+        $form = $companyFormHelper->getCompanyForm($company);
+
+        if ($request->isMethod('POST')) {
+            $company->setFromArray($request->request->get($form->getName()));
+
+            $form->handleRequest($request);
+
+            if ($form->isValid()) {
+                $files = $request->files->get($form->getName());
+                $path = 'images/company/';
+
+                if ($files['FileUpload'] != null ){
+                    $filename = $files['FileUpload']->getClientOriginalName();
+                    $files['FileUpload']->move($path, $filename);
+                    $company->logo_src = $filename;
+                } else {
+                   $company->logo_src = $company->logo_src;
+                }
+
+                if (null === $id) {
+                    $app['dbs']['mysql_write']->insert('company', $company->toArray());
+                    $id = $app['dbs']['mysql_write']->lastInsertId();
+                } else {
+                    $company->id = $id;
```

Comment:
Here `$company` should already have the `$id`, so this line can be deleted. But please test it.